### PR TITLE
Remove global defs

### PR DESCRIPTION
**Pull request by beyang**
This removes the `GlobalDefs` store and adds instrumentation to its replacement, the `Defs` store.

After this is merged, we should be able to drop the `global_defs` table.

### CODE REVIEW

File: services/backend/internal/localstore/defs.go [hunk 1]
```diff
@@ -402,11 +402,11 @@ func (s *defs) UpdateFromSrclibStore(ctx context.Context, op store.DefUpdateOp)
 	if err := accesscontrol.VerifyUserHasWriteAccess(ctx, "Defs.UpdateFromSrclibStore", op.Repo); err != nil {
 		return err
 	}
-
 	repo, err := store.ReposFromContext(ctx).Get(ctx, op.Repo)
 	if err != nil {
 		return err
 	}
+	__observer := newDefsUpdateObserver("defs2", repo.URI)
```
Comment (keegancsmith) [inline services/backend/internal/localstore/defs.go]: why the funky name? (`__` prefix)

File: services/backend/internal/localstore/defs.go [hunk 2]
```diff
@@ -439,6 +441,10 @@ func (s *defs) Update(ctx context.Context, op store.DefUpdateOp) error {
 		return err
 	}
 
+	__observer := newDefsUpdateObserver("defs2", repo.URI)
+	__observer.start("update_total")
```
Comment (keegancsmith) [inline services/backend/internal/localstore/defs.go]: this api is pretty tricky. I much smoother to use API would just be getting start to return a `func()` which records the end time. That way you don't have to remember to use the same string / prevents typos causing runtime errors.

File: services/backend/internal/localstore/defs.go (hunk 1 above)
Comment (keegancsmith) [inline services/backend/internal/localstore/defs.go]: also there is nothing measuring the total time for `UpdateFromSrclibStore`

File: services/backend/internal/localstore/defs.go [hunk 3]
```diff
@@ -467,17 +473,22 @@ func (s *defs) Update(ctx context.Context, op store.DefUpdateOp) error {
 	dbh := graphDBH(ctx)
 
 	// Update def_keys
+	__observer.start("update def_keys")
 	if err := updateDefKeys(ctx, dbh, repo.URI, chosenDefs); err != nil {
+		__observer.end("update def_keys")
 		return err
 	}
+	__observer.end("update def_keys")
```
Comment (keegancsmith) [inline services/backend/internal/localstore/defs.go]: nit: I don't think it is enforced, but generally prometheus labels I have seen don't contains whitespace

File: services/backend/internal/localstore/defs_helpers.go [hunk 4]
```diff
@@ -1,44 +1,74 @@
...
+		Name:      "search_duration_seconds",
+		Help:      "Duration for def search",
+		MaxAge:    time.Hour,
+	}, []string{"table", "repo", "part"})
+	prometheus.MustRegister(defsSearchDuration)
 }
 
-type defsUpdateObserver struct {
+type observer struct {
+	mu          sync.Mutex
```
Comment (keegancsmith) [inline services/backend/internal/localstore/defs_helpers.go]: if you use the API I mentioned earlier of just returning a closure for end, you can avoid keeping track of the starts map and this mutex.

**Comment by keegancsmith** [comment]
Some inline comments, but otherwise LGTM

**Comment by keegancsmith** [comment]
@beyang I want to merge this in ASAP since it will help the async queue move faster. So I did my suggestions in https://github.com/sourcegraph/sourcegraph/pull/185/commits/b2e982e877acd3083229bb2f7cb4a3aec1f2cbf2
If that looks good to you I'll merge.

**Comment by beyang** [comment]
LGTM thanks!

**Pull request merged**